pkg/sync/sync.go

```diff
@@ -887,3 +894,112 @@ func CloneStatus(engineControllerClient *client.ControllerClient) (map[string]*S
 	}
 	return cloneStatusMap, nil
 }
+
+func (t *Task) HashSnapshot(snapshotNames []string, rehash bool, concurrentLimit int) error {
+	replicas, err := t.client.ReplicaList()
+	if err != nil {
+		return err
+	}
+
+	for _, r := range replicas {
+		if r.Mode != types.RW {
+			return fmt.Errorf("cannot hash a snapshot because %v is in %v mode", r.Address, r.Mode)
```

Comment:
I am thinking if the cmd should error out when there is a non-RW replica...

Reply:
If the replica is in WO state, the data might be changed. Do you think it should no error out here?

Reply:
I mean, postponing the hashing. If it makes thing complicated, we can ignore this now.

Reply:
Maybe requeue it in snapshot monitor?

Reply:
Implement the requeue the hash request in longhorn-manager 'ssnapshot monitor if encounter this error.